### PR TITLE
Added unit tests for the diagonal function

### PR DESCRIPTION
**Pull request by sakchal**
This PR introduces a suite of unit tests for lower.py located within the within the create_array directory. The primary functions in this file facilitate the creation of an an array with a specified diagonal, where this specified diagonal is passed in as an AFArray. 

Diagonal: Ensures that the diagonal of the new array matches the shape and the value of of the original array given.
Diagonal Invalid Shape: Ensures that proper error checking occurs if a diagonal array is trying to be constructed from an array with incorrect dimensions (can only be 1D or 2D).

### CODE REVIEW

File: tests/test_diag.py [hunk 1]
```diff
@@ -0,0 +1,50 @@
+import pytest
+
+import arrayfire_wrapper.dtypes as dtypes
+import arrayfire_wrapper.lib as wrapper
+
+# tests to include
+# - if diagonal created is same shape as the input diagonal
+# - if diagonal value is same value as input diagonal
+# - if invalid diagonal is passed in
```
Comment (roaffix) [inline tests/test_diag.py:9]: Is this comment necessary here?

Comment (sakchal) [reply tests/test_diag.py]: deleted them in the new commit

**Comment by sakchal** [comment]
@roaffix getting black and other formatting errors in manage_array.py. Should I go in and fix that file's formatting or leave it?

**Comment by roaffix** [comment]
> @roaffix getting black and other formatting errors in manage_array.py. Should I go in and fix that file's formatting or leave it?

Its fine. I'll fix them all in some time

**Review by roaffix — APPROVED** [review]

**Pull request merged**